internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -279,31 +335,163 @@ func resourceMonitorDiagnosticSettingCreateUpdate(d *pluginsdk.ResourceData, met
 	}
 
 	if v := d.Get("log_analytics_destination_type").(string); v != "" {
-		if workspaceId != "" {
-			parameters.Properties.LogAnalyticsDestinationType = &v
-		} else {
-			return fmt.Errorf("`log_analytics_workspace_id` must be set for `log_analytics_destination_type` to be used")
-		}
+		parameters.Properties.LogAnalyticsDestinationType = &v
 	}
 
 	if !valid {
 		return fmt.Errorf("either a `eventhub_authorization_rule_id`, `log_analytics_workspace_id`, `partner_solution_id` or `storage_account_id` must be set")
 	}
 
-	if _, err := client.CreateOrUpdate(ctx, diagnosticSettingId, parameters); err != nil {
-		return fmt.Errorf("creating Monitor Diagnostics Setting %q for Resource %q: %+v", name, actualResourceId, err)
+	if _, err := client.CreateOrUpdate(ctx, id, parameters); err != nil {
+		return fmt.Errorf("creating Monitor Diagnostics Setting %q for Resource %q: %+v", id.Name, id.ResourceUri, err)
 	}
 
-	read, err := client.Get(ctx, diagnosticSettingId)
+	d.SetId(fmt.Sprintf("%s|%s", id.ResourceUri, id.Name))
```

Comment:
.. so perhaps we should be assigning this above into a local variable to be able to reuse it?

```suggestion
	resourceId := fmt.Sprintf("%s|%s", id.ResourceUri, id.Name)
	// ...
	d.SetId(resourceId)
```